### PR TITLE
xmlrunner: Expose python docstrings as comments in the XML result file

### PR DESCRIPTION
**Pull request by oliviermartin**
These comments could for instance be retrieved by XSLT to add
description to test title.

### CODE REVIEW

File: xmlrunner/result.py [hunk 1]
```diff
@@ -258,6 +261,11 @@ def startTest(self, test):
                 # Handle partial and partialmethod objects.
                 test_method = getattr(test_method, 'func', test_method)
                 _, self.lineno = inspect.getsourcelines(test_method)
+
+                if test_method.__doc__:
+                    self.doc = test_method.__doc__
+                else:
+                    self.doc = None
```
Comment (dnozay) [inline xmlrunner/result.py]: isn't this sufficient then?
```
self.doc = test_method.__doc__
```

File: xmlrunner/result.py [hunk 2]
```diff
@@ -555,6 +563,9 @@ def _report_testcase(test_result, xml_testsuite, xml_document):
         if test_result.lineno is not None:
             testcase.setAttribute('line', str(test_result.lineno))
 
+        if test_result.doc is not None:
+            testcase.appendChild(xml_document.createComment(str(test_result.doc)))
```
Comment (dnozay) [inline xmlrunner/result.py]: does this escape comments closing tags?

Comment (oliviermartin) [reply xmlrunner/result.py]: Yes, it does.

**Comment by oliviermartin** [comment]
> please add positive / negative testcases; and in particular, please cover cases where `__doc__` looks like xml.

I added the positive/negative testcases.

As far as I can see the build fails but it is due to `HEAD` of `master` being broken.

File: xmlrunner/result.py [hunk 3]
```diff
@@ -555,6 +560,12 @@ def _report_testcase(test_result, xml_testsuite, xml_document):
         if test_result.lineno is not None:
             testcase.setAttribute('line', str(test_result.lineno))
 
+        if test_result.doc is not None:
+            comment = str(test_result.doc)
+            # The use of '--' is forbidden in XML comments
+            comment = comment.replace('--', '- -')
```
Comment (dnozay) [inline xmlrunner/result.py]: nit: can we drop the space and use `&#45;` for the second `-`?

**Pull request merged**